jwql/shared_tasks/shared_tasks.py

```diff
@@ -224,9 +224,43 @@ def collect_after_task(**kwargs):
     gc.collect()
 
 
-def run_subprocess(name, cmd, outputs, cal_dir, ins, in_file, short_name, res_file, cores):
-    command = "{} {} {} '{}' {} {} {} {}"
-    command = command.format(name, cmd, outputs, cal_dir, ins, in_file, short_name, cores)
+def convert_step_args_to_string(args_dict):
```

Comment:
Actually, looking at this function, I have what might be a naive question – if we're using `json.loads()` to parse the command-line argument, is there a reason we're not using `json.dumps()` to encode it?

Reply:
Good question. I can compare the output from that and remove this function if it's not needed.

Reply:
Looks like `json.loads()` works. I'll remove this function and switch to using that.

Reply:
I take that back. The previous results were deceiving because I hadn't restarted the celery worker. When I do that, `json.dumps()` does not appear to work for translating the dictionary to the necessary string.

Reply:
That's annoying. Also, it kind of feels like a python issue if taking a dictionary, turning it into a string with `json.dumps()`, and turning it back into a dictionary with `json.loads()` *doesn't* give back the same dictionary.

Reply:
I'm starting with the dictionary:
step_args = {"dark_current": {"skip":True}}

json.dumps turns that into: 
'{"dark_current": {"skip": true}}'

But then in the command that is pieced together to call the pipeline, it looks like that is being broken up into multiple strings:
'--step_args', '{dark_current:', '{skip:', 'true}}'

I might just go back to the function I had written, which was doing the conversion correctly.

Reply:
Looking at the file, it seems as if, in `run_subprocess()`, you're calling `json.loads()` on the dictionary, which seems as if it might the the opposite of what you want to be doing?

Reply:
Ignore that. That's a typo. I've been testing on the server using json.dumps()